hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SubGraphTraverser.java

```diff
@@ -64,41 +68,61 @@ public SubGraphTraverser(HugeGraph graph) {
         checkLimit(limit);
 
         Id labelId = this.getEdgeLabelId(label);
-        Traverser traverser = new Traverser(sourceV, labelId, degree,
-                                            capacity, limit, rings);
+        Traverser traverser = new Traverser(sourceV, labelId, depth, degree,
+                                            capacity, limit, rings,
+                                            sourceInRing);
         List<Path> paths = new ArrayList<>();
         while (true) {
             paths.addAll(traverser.forward(dir));
-            boolean reachDepth = rings ? --depth <= 0 : depth-- <= 0;
-            if (reachDepth || traverser.reachLimit() ||
+            if (--depth <= 0 || traverser.reachLimit() ||
                 traverser.finished()) {
                 break;
             }
         }
         return paths;
     }
 
+    private static boolean hasMultiEdges(List<Edge> edges, Id target) {
+        int count = 0;
+        for (Edge edge : edges) {
+            if (((HugeEdge) edge).id().otherVertexId().equals(target)) {
+                if (++count > 1) {
+                    return true;
+                }
+            }
+        }
+        assert count == 1;
```

Comment:
seems count maybe = 0

Reply:
there already exists one edge connected to 'target', so count must be >= 1, false means count == 1